### ISSUE TITLE
[M13] N4: Prefill via parallel scan in the Swift engine

### ISSUE THREAD

**Issue by travisgalloway** [travisgalloway/monica#169]
Part of #163.

Carry the parallel‑scan prefill into the Swift engine so the prompt is ingested in one pass instead of token‑by‑token — the biggest single‑stream latency win.

## Scope
- Implement `prefill(tokens) -> (logits, State)` in Swift, mirroring the Python seam method (#165): run the SSD chunked scan + batched attention over the prompt, extract the recurrent `State` (Mamba conv window + final ssm_state `new_states[:, -1]` read at L‑1; attention K/V caches; MoE placeholder), then hand off to the mlx‑swift‑lm decode loop.

## Acceptance criteria
- Swift `prefill` + decode == Swift pure step‑by‑step decode, fp32 ~1e‑4.
- Swift `prefill` state/logits == the Python engine's `prefill` (#165) on the same prompt/weights, fp32 ~1e‑4.
- Measured prefill latency win vs the token‑by‑token baseline (feeds the bench, #N5).

## Dependencies
Blocked by #166 (N1 port) and #165 (Python prefill seam / parity oracle).

**Comment by travisgalloway** [comment]
Implemented and PR opened: #291

- `MonicaModel.prefill(tokens, lastOnly:)` — one-shot SSD parallel scan over the prompt (`SelectiveSSM.parallelWithState`, `convWindow`, `forwardPrefill` on each block type), wired into `Generator.generate` (default `usePrefill: true`; `usePrefill: false` / `--no-prefill` keeps the old sequential path as the AC3 baseline and rollback).
- `monica-parity` P1-P5: forward-vs-prefill logits, `lastOnly` honesty, **state handoff verified two independent ways** (cross-language vs a new `prefill.safetensors` Python oracle, and intra-Swift vs `step`'s own recurrence), prefill-then-decode vs pure step-by-step, and exact greedy-id equality through the prefill path. fp32 `rtol=1e-4`/`atol=1e-5` unchanged.
- New `toy-short` fixture covers the conv-window left-pad edge case (`L < d_conv-1`) no existing fixture reaches.
- `monica-generate --bench-prefill` (AC3) wired into CI as an informational, non-gating step.

Locally verified: full pytest suite (1572 passed), the extended fixture-staleness test, the smoke gate, and `swift build`/`--self-test` for both `swift/` and `swift/engine/`. The Swift-side numeric gate (`monica-parity`, `xcodebuild`) is CI-only on this host — no Xcode, confirmed pre-existing. See PR #291 for full detail.

**Comment by travisgalloway** [comment]
#170's harness recorded the AC3 first-green-CI numbers into \`docs/design/14-inference-engine.md\` and \`docs/benchmarks.md\`:

\`\`\`
bench-prefill: prompt_len=128 iterations=5  sequential=332.62ms  parallel-scan=6.01ms  speedup=55.30x
bench-prefill: sequential argmax=186  parallel-scan argmax=186
\`\`\`

From the green \`swift-engine\` CI run [31777284815](https://github.com/travisgalloway/monica/actions/runs/31777284815) on \`main\`, \`Fixtures/toy\` — the hosted macOS runner, explicitly **not** a local-hardware measurement. \`monica-bench --mode prefill\` (#170) now calls the same \`Bench.prefill\` function \`--bench-prefill\` calls, so this figure remains the comparable baseline going forward rather than being orphaned.
